server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -230,6 +230,16 @@ public long relativeTimeInMillis() {
         return cachedTimeThread.relativeTimeInMillis();
     }
 
+    /**
+     * Returns a value of nanoseconds that may be used for relative time calculations.
+     *
+     * This method should only be used for calculating time deltas. For an epoch based
+     * timestamp, see {@link #absoluteTimeInMillis()}.
+     */
+    public long relativeTimeInNanos() {
+        return TimeValue.msecToNSec(cachedTimeThread.relativeTimeInMillis());
```

Comment:
@tbrooks8 I'm not sure we should add this. The underlying logic as it is coded up now isn't nano second precise.
We could adjust `CachedTimeThread` to return nano second precision times and only add the rounding to `ms` in the existing `relativeTimeInMillis` method?
But then again, maybe we don't need `ns` for the http timeout in the first place and can just work with `ms` precision and not add more complication to the `ThreadPool` class?

Reply:
> We could adjust CachedTimeThread to return nano second precision times and only add the rounding to ms in the existing relativeTimeInMillis method?

I think this is the correct approach, but I hesitated to change it in this PR as the behavior of the timer tick thread might be of concern to other people. Anyway, I will make that change and we can discuss.

Reply:
Anyway, I made this change.

1. We need nanoseconds. `TaskScheduler` deals in nanoseconds.
2. If we do the conversion outside of the `ThreadPool` we depend on the fact that `ThreadPool` will always return a relative millis value that can be successfully converted to milliseconds. Obviously this is the case, but I prefer to not depend on that fact. Which is why I prefer that `ThreadPool` returns nanotime (similar to the fact that most Java relative time calculations are made using nanotime). 
3. The cost of the change of division every time you want millis. I assume this is fine, but obviously the entire point of this tick thread is to replace a `System.nanotime` call with a volatile read at the cost of precision. So maybe there is some objection. We could also store a precalculated version in millis.